mflike/foreground.py

```diff
@@ -482,36 +640,271 @@ def _bandpass_construction(self, _initialize=False, **params):
+                        trans = bp / trans_norm * _cmb2bb(nub)
+                        if "tt" in self.requested_cls or "te" in self.requested_cls:
+                            self.bandint_freqs_T.append([nub, trans])
+                        if "te" in self.requested_cls or "ee" in self.requested_cls:
+                            self.bandint_freqs_P.append([nub, trans])
+                    else:
+                        if self.bandsh_beams_path:
+                            blT, blP = self.return_beams(exp, nu_ghz, shift)
+                        else:
+                            # not propagating bandpass shifts to the chromatic beams
+                            blT, blP = self.return_beams(exp, nu_ghz, 0.)
+
+                        if "tt" in self.requested_cls or "te" in self.requested_cls:
+                            trans_normT = trapezoid(
+                                bp[..., np.newaxis] * _cmb2bb(nub)[..., np.newaxis] * blT, nub, axis=0
+                            )
+                            ratioT = (
+                                bp[..., np.newaxis] * _cmb2bb(nub)[..., np.newaxis] * blT / trans_normT
+                            )
+                            self.bandint_freqs_T.append([nub, ratioT])
+
+                        if "te" in self.requested_cls or "ee" in self.requested_cls:
+                            trans_normP = trapezoid(
+                                bp[..., np.newaxis] *  _cmb2bb(nub)[..., np.newaxis] * blP, nub, axis=0
+                            )
+                            ratioP = (
+                                bp[..., np.newaxis] * _cmb2bb(nub)[..., np.newaxis] * blP / trans_normP
+                            )
+                            self.bandint_freqs_P.append([nub, ratioP])
 
         # fgspectra can't mix monofrequency with [nu, bp]. If one channel is mono-frequency then we
         # assume all of them and pass to fgspectra an array (not list!!) of frequencies
         if data_are_monofreq:
-            self.bandint_freqs = np.asarray(self.bandint_freqs)
+            self.bandint_freqs_T = np.asarray(self.bandint_freqs_T)
+            self.bandint_freqs_P = np.asarray(self.bandint_freqs_P)
             if self._initialized:
                 self.log.info("bandpass is delta function, no band integration performed")
         self._initialized = True
 
+    ###########################################################################
+    ## This part deals with beam functions, i.e. reading beam from file or
+    ## computing it as a Gaussian beam. We also have a function to compute
+    ## the correction expected for a Gaussian beam in case of bandpass shift
+    ## that should be applied to any beam profile
+    ###########################################################################
+    def _read_yaml_file(self, file_path):
+        
+        from cobaya.yaml import yaml_load_file
+        
+        filename = "%s.yaml" % file_path
```

Comment:
Sorry I didn't get it at the beginning but the `Bandpass_shifted_beams` is actually the basename of the `yaml` file so you have to add the `.yaml` extension. I think it is a bit misleading and you can always have someone (like me) that use `.yml` extension file. It's up to you but I'd prefer if the `Bandpass_shifted_beams` is really the path to the `yaml` file and not something like `/where/is/the/yaml/file/bandpass_shifted_beams`. This way you can remove this internal method `_read_yaml_file` and directly rely to `cobaya's yaml_load_file` function